### PR TITLE
[Messenger] Make sure Sender and Receiver locators have valid services

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 4.1
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

This make sure that Sender and Receiver locators have valid services. Also adds minor improvements into consume command.

### CODE REVIEW

File: src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php [hunk 1]
```diff
@@ -87,12 +87,10 @@ protected function configure(): void
     protected function execute(InputInterface $input, OutputInterface $output): void
     {
         if (!$this->receiverLocator->has($receiverName = $input->getArgument('receiver'))) {
-            throw new \RuntimeException(sprintf('Receiver "%s" does not exist.', $receiverName));
+            throw new RuntimeException(sprintf('Receiver "%s" does not exists.', $receiverName));
```
Comment (ogizanagi) [inline src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php]: That's not a typo. Receiver **does** or **doesn't** exist :)

**Review by ogizanagi — APPROVED** [review]

File: src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php (hunk 1 above)
Comment (yceruto) [reply src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php]: Reverted, thanks! :smile:

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 2]
```diff
@@ -162,6 +164,11 @@ private function registerReceivers(ContainerBuilder $container)
     {
         $receiverMapping = array();
         foreach ($container->findTaggedServiceIds($this->receiverTag) as $id => $tags) {
+            $receiverClass = $container->findDefinition($id)->getClass();
+            if (!is_subclass_of($receiverClass, ReceiverInterface::class)) {
+                throw new RuntimeException(sprintf('Receiver class "%s" must implement interface "%s".', $receiverClass, ReceiverInterface::class));
```
Comment (sroze) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: Can you add more context to the exception? I suggest:
```
Invalid receiver "{id}": class "{class}" must implement interface "{interface}"
```

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 3]
```diff
@@ -178,6 +185,11 @@ private function registerSenders(ContainerBuilder $container)
     {
         $senderLocatorMapping = array();
         foreach ($container->findTaggedServiceIds($this->senderTag) as $id => $tags) {
+            $senderClass = $container->findDefinition($id)->getClass();
+            if (!is_subclass_of($senderClass, SenderInterface::class)) {
+                throw new RuntimeException(sprintf('Sender class "%s" must implement interface "%s".', $senderClass, SenderInterface::class));
```
Comment (sroze) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: Same for here.

File: src/Symfony/Component/Messenger/Tests/DependencyInjection/MessengerPassTest.php [hunk 4]
```diff
@@ -330,6 +356,14 @@ public function stop(): void
     }
 }
 
+class InvalidDummyReceiver
```
Comment (sroze) [inline src/Symfony/Component/Messenger/Tests/DependencyInjection/MessengerPassTest.php]: You can probably remove `Dummy` in here 🙃

**Review by sroze — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php (hunk 2 above)
Comment (yceruto) [reply src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: Better, thanks.

File: src/Symfony/Component/Messenger/Tests/DependencyInjection/MessengerPassTest.php (hunk 4 above)
Comment (yceruto) [reply src/Symfony/Component/Messenger/Tests/DependencyInjection/MessengerPassTest.php]: Removed.

**Comment by yceruto** [comment]
Status: Needs Review

**Review by sroze — APPROVED** [review]

**Comment by yceruto** [comment]
Rebased.

**Comment by sroze** [comment]
Thank you @yceruto.

**Pull request merged**